packages/adapter-auto/index.js

```diff
@@ -10,31 +13,39 @@ for (const candidate of adapters) {
 
 		try {
 			module = await import(candidate.module);
```

Comment:
am suddenly wondering if we should actually be importing the module from the app directory rather than the adapter-auto directory. in most cases it should Just Work, but you can imagine a situation where `adapter-auto` is installed in the workspace root while `adapter-foo` is installed inside the app

Reply:
Doesn't `candidate.module` just contain the name of the package, and the resolution algorithm should start at the `adapter-auto` directory and look for node_modules there, and if it's there, look for that module there? At least I (try to) use this fact by installing the package into the `adapter-auto` directory.

Reply:
If the adapter is a dependency of the app (which it would be, if you'd already installed it) then surely resolution should start from there? (Until `import.meta.resolve` is stable, this would need [import-meta-resolve](https://www.npmjs.com/package/import-meta-resolve))

Reply:
I feel like this overcomplicates things for no good reason. In what world are you using adapter-auto, but have installed a more specific adapter in another place? You either have them in the same place or switched to the one you actually want to use a long time ago.

Reply:
It's not so hard to imagine a situation where shared dependencies are installed in the workspace root but someone installs a one-off dependency in a package — this very repo used to work that way, until we decided to move all dependencies into packages. I also wonder if 'dependencies of `x` can import all other dependencies of `x`' is guaranteed across all package managers now and in the future. It's a side-effect of the resolution algorithm plus the way package managers populate `node_modules`, but it's the sort of thing that feels changeable, the same way pnpm prevents the npm 3+ behaviour of allowing `x` to directly import all indirect dependencies